client/base_client.go

```diff
@@ -13,4 +14,6 @@ type BaseClient interface {
 		headers map[string]interface{}, body ...byte) (*http.Response, error)
 	SetOauth(auth OAuth)
 	OAuth() OAuth
+	SendRequestWithCtx(ctx context.Context, method string, rawURL string, data url.Values,
```

Comment:
```suggestion
	SendRequestWithContext(ctx context.Context, method string, rawURL string, data url.Values,
```

Pretty much everywhere in Go land spells this whole thing out

Reply:
Addressed